src/celeritas/field/FieldDriver.hh

```diff
@@ -370,27 +370,14 @@ FieldDriver<StepperT>::one_good_step(real_type step, OdeState const& state) cons
     // Update state, step taken by this trial and the next predicted step
     output.end.state = result.end_state;
     output.end.step = step;
-    output.proposed_step = (errmax2 > ipow<2>(options_.errcon))
-                               ? options_.safety * step
-                                     * fastpow(errmax2, half() * options_.pgrow)
-                               : options_.max_stepping_increase * step;
+    output.proposed_step
```

Comment:
Even though errcon is an arbitrary number, what is the rationale dropping errcon from the original code as it changes the logic?

Reply:
The exactly analogous error calculation in "one good step" used a different criteria with the same value: max rel. I thought it might've been an oversight that these two different values existed.

Reply:
There are a few number of parameters for adaptive step controls (in Geant4 even though some of them are empirically driven), delta-chord, relative error max, power of grow or shrink for the next optimal stride, epsilon and so on.  As far as I know, the `errcon` is one of parameters from the original numerical recipe in c (epsilon or errmax) and a key parameter to control efficiency or quality of the step size control.  I am not a big fan for many fuzzy parameters or magic algorithms, but support whatever if is right and efficient if there is a good logical argument why it should be.  On the other hand, we should also balance to copy the current Geant4 field propagation and driver workflow and to adapt alternative algorithms as much as possible for SIMT.  Nevertheless, the goal is to develop accurate and efficient field driver (probably the simpler, the better) as there is no de-facto standard for the adaptive step size control (not like steppers themself) other than stability or accuracy tests with a uniform magnetic field.

Reply:
I'm not opposed to having these parameters, but it really *looks* like there are a bunch of inconsistencies in how the errors are calculated. Let me try reverting all the "fixes" so this becomes a simple refactor that better highlights the inconsistencies I'm seeing.